Package.swift

```diff
@@ -371,3 +385,23 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(path: "../swift-crypto"),
     ]
 }
+
+#if compiler(>=5.5)
+// SwiftSyntax depends on lib_InternalSwiftSyntaxParser from the toolchain,
+// which had an ABI break in Swift 5.5. As a result, we shouldn't attempt to
+// compile PackageSyntax with an earlier compiler version. Although PackageSyntax
+// should compile with any 5.5 compiler, it will only be functional when built
+// with a toolchain that has a compatible parser library.
+if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
+    package.dependencies += [.package(url: "https://github.com/apple/swift-syntax.git", .branch(relatedDependenciesBranch))]
```

Comment:
Should this be pinned to a particular release of SwiftSyntax rather than the branch?  The toolchain build will use the branch, but it will presumably also set `SWIFTCI_USE_LOCAL_DEPS`.

Reply:
I think we should use the branch here - it ensures we only ever need to support one version of the SwiftSyntax API, which changes pretty often

Reply:
But if it changes often, doesn't that mean that changes to the branch could suddenly break SwiftPM?  i.e. would it not make sense to specify a semantic version and then when necessary update SwiftPM to use the newer version at the same time as adopting any necessary changes, in a single PR?

Reply:
I believe SwiftSyntax CI builds SwiftPM so I think we'd at least catch breaking changes pre-merge. The main problem I'm running into is that SwiftPM always needs to build against top-of-tree SwiftSyntax for the feature to work in nightly toolchains. We could also support a specific tag for local development, but it wouldn't be guaranteed to compile 100% of the time